### PR TITLE
Code fix for the bug in list by immediate parent

### PR DESCRIPTION
**Pull request by sarangan12**
@dsgouda @vishrutshah @veronicagg @salameer Please review and approve

### CODE REVIEW

File: src/modeler/AutoRest.Swagger/Validation/Core/JsonValidationLogListener.cs [hunk 1]
```diff
@@ -6,12 +6,14 @@
...
 using System.Collections.Generic;
 using System.Linq;
+using System.Text.RegularExpressions;
 
 namespace AutoRest.Swagger.Logging.Core
 {
     public class JsonValidationLogListener : ILogListener
     {
         private readonly List<Dictionary<string, string>> rawMessageCollection = new List<Dictionary<string, string>>();
+        public static readonly Regex ResourcePathPattern = new Regex(@"/providers/(?<providerNamespace>[^{/]+)((/(?<resource>[^{/]+)/)((?<resourceName>[^/]+)))+(/(?<unparameterizedresource>[^{/]+))?");
```
Comment (dsgouda) [inline src/modeler/AutoRest.Swagger/Validation/Core/JsonValidationLogListener.cs]: nit: lets keep this private

File: src/modeler/AutoRest.Swagger/Validation/TrackedResourceListByImmediateParent.cs [hunk 2]
```diff
@@ -35,23 +35,83 @@ public class TrackedResourceListByImmediateParent : TypedRule<Dictionary<string,
...
+        /// <summary>
+        /// Gets the immediate parent resource name
+        /// </summary>
+        /// <param name="childResourcePaths">paths that fits the child resource criteria</param>
+        /// <returns>name of the immediate parent resource name</returns>
+        private string GetImmediateParentResourceName(string pathToEvaluate)
+        {
+            pathToEvaluate = pathToEvaluate.Substring(0, pathToEvaluate.LastIndexOf("/{"));
+            pathToEvaluate = pathToEvaluate.Substring(0, pathToEvaluate.LastIndexOf("/{"));
+            return pathToEvaluate.Substring(pathToEvaluate.LastIndexOf("/") + 1);
```
Comment (dsgouda) [inline src/modeler/AutoRest.Swagger/Validation/TrackedResourceListByImmediateParent.cs]: Could you explain what happens if there is only child resource in the path?

File: src/modeler/AutoRest.Swagger/Validation/Core/JsonValidationLogListener.cs (hunk 1 above)
Comment (sarangan12) [inline src/modeler/AutoRest.Swagger/Validation/Core/JsonValidationLogListener.cs]: Done

File: src/modeler/AutoRest.Swagger/Validation/TrackedResourceListByImmediateParent.cs (hunk 2 above)
Comment (sarangan12) [inline src/modeler/AutoRest.Swagger/Validation/TrackedResourceListByImmediateParent.cs]: @dsgouda  What do you mean by "if there is only child resource in the path"? Can you give an example?

Comment (dsgouda) [inline src/modeler/AutoRest.Swagger/Validation/TrackedResourceListByImmediateParent.cs]: ```/subscriptions/subscriptionId/resourceGroups/{resourceGroupName}/providers/SampleProviderNamespace/OnlyChildResource/{onlyChildResourceName}/```

**Pull request merged**